examples/30_multi_dimensional_cp_ct.py

```diff
@@ -0,0 +1,101 @@
+# License for the specific language governing permissions and limitations under
+# the License.
+
+# See https://floris.readthedocs.io for documentation
+
+
+import numpy as np
+
+from floris.tools import FlorisInterface
+
+
+"""
+This example follows the same setup as example 01 to createa a FLORIS instance and:
+1) Makes a two-turbine layout
+2) Demonstrates single ws/wd simulations
+3) Demonstrates mulitple ws/wd simulations
+
+with the modification of using a turbine definition that has a multi-dimensional Cp/Ct table.
+
+In the input file `gch_multi_dim_cp_ct.yaml`, the turbine_type points to a turbine definition,
+iea_15MW_floating_multi_dim_cp_ct.yaml located in the turbine_library,
+that supplies a multi-dimensional Cp/Ct data file in the form of a .csv file. This .csv file
+contains two additional conditions to define Cp/Ct values for: Tp for wave period, and Hs for wave
+height. For every combination of Tp and Hs defined, a Cp/Ct/Wind speed table of values is also
+defined. It is required for this .csv file to have the last 3 columns be ws, Cp, and Ct. In order
+for this table to be used, the flag 'multi_dimensional_cp_ct' must be present and set to true in
+the turbine definition. Also of note is the 'velocity_model' must be set to 'multidim_cp_ct' in
+the main input file. With both of these values provided, the solver will downselect to use the
+interpolant defined at the closest conditions. The user must supply these conditions in the
+main input file under the 'flow_field' section, e.g.:
+
+flow_field:
+  multidim_conditions:
+    Tp: 2.5
+    Hs: 3.01
+
+The solver will then use the nearest-neighbor interpolant. These conditions are currently global
+and used to select the interpolant at each turbine.
+
+Also note in the example below that there is a specific method for computing powers when
+using turbines with multi-dimensional Cp/Ct data under FlorisInterface, called
+'get_turbine_powers_multidim'. The normal 'get_turbine_powers' method will not work.
+"""
+
+# Initialize FLORIS with the given input file via FlorisInterface.
+fi = FlorisInterface("inputs/gch_multi_dim_cp_ct.yaml")
+
+# Convert to a simple two turbine layout
+fi.reinitialize(layout_x=[0., 500.], layout_y=[0., 0.])
+
+# Single wind speed and wind direction
+print('\n========================= Single Wind Direction and Wind Speed =========================')
+
+# Get the turbine powers assuming 1 wind speed and 1 wind direction
+fi.reinitialize(wind_directions=[270.], wind_speeds=[8.0])
+
+# Set the yaw angles to 0
+yaw_angles = np.zeros([1,1,2]) # 1 wind direction, 1 wind speed, 2 turbines
+fi.calculate_wake(yaw_angles=yaw_angles)
+
+# Get the turbine powers
+turbine_powers = fi.get_turbine_powers_multidim()/1000.
+# turbine_powers = fi.get_turbine_powers()/1000.
+print('The turbine power matrix should be of dimensions 1 WD X 1 WS X 2 Turbines')
+print(turbine_powers)
+print("Shape: ",turbine_powers.shape)
+
+# Single wind speed and wind direction
```

Comment:
Comment should be updated

Reply:
Updated! And updated example 01 with the same.